accepted/2021/package-scoring/PackageScoring.md

```diff
@@ -0,0 +1,259 @@
+    - For diamond dependencies, a minor/patch version isn’t as breaking.
+  - “Harm reduction” is an idea that people are going to do risky activities or be in risky situations and instead of saying “don’t be risky”, we can choose to help make those activities less risky by being proactive.
+    - Let’s make maintainer’s lives easier to respond to and fix issues while encouraging community contribution.
+- **Community**
+  - Make it easier for someone to use a package with an abundance of documentation.
+  - Clearly explain to someone how they might contribute to a package they use.
+  - Demonstrate to people that packages are impactful, valuable, and have traction in the ecosystem.
+  - Help package authors sustain their work by incentivizing their time with raising donations/funding.
+
+There are also many **jobs-to-be done** when browsing for packages to use in projects or deciding to keep in your software supply chain. Here is a small list of jobs this concept will help developers with.
+
+1. Determine whether a package is recognized by the ecosystem based on its **popularity and community** score.
+2. Understand if a package is actively maintained or abandonded based on its **maintenance, quality, and community** score.
+3. Prevent installing a package that contains a known vulnerability that has not yet been resolved based on it's **maintenance and quality** score.
+4. See how they can get involved with the development of the package based on its **community** score.
+5. Assess whether the package & its dependencies are compatible with the latest .NET Frameworks & tooling based on it's **quality** score.
+
+The creation of a score for each of these categories and an overall score per package will help assess the health of the .NET Ecosystem & NuGet.
+
+A developer for example can assess the score of a package to make a better decision as to if the package is worth including in their project. A team at a company might even take this score into their dependency policy to say they only accept a package score of 80 or above because they have stricter requirements.
+
+## Concepts
+
+**NuGet Gallery**
+
+![](PackageScoring.png)
+
+![](PackageScoringList.png)
+
+**Visual Studio**
+
+![](PackageScoringVisualStudio.png)
+
+**.NET CLI / Tool**
+
+![](PackageScoringTool.png)
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+When a .NET developer creates a NuGet package and pushes it to NuGet.org, it will be scored on the following metrics totaling 25 points per category up to an overall score of 100.
+
+**Popularity**:
+
+- Total Weekly Downloads
+- Number of Dependents
+- Total GitHub Stars
+- Total GitHub Forks
+- Total GitHub Contributors
```

Comment:
Would this force a project to be on Github to have a good rep?

Reply:
Not likely. Just a healthy amount of stars/forks/contributors to determine the optimal stop to achieve the maximum score.

Reply:
> Not likely. Just a healthy amount of stars/forks/contributors to determine the optimal stop to achieve the maximum score.

I think the question was: if stars/forks/contributors are considered, how will the package score be affected for packages that aren't on GitHub, e.g. because are on Bitbucket or a proprietary?

Reply:
I don't have a great answer. It's an unresolved question at the bottom for now. 

The two main paths I would see is we would use the information we get from a connected repository provider as an additive to the popularity score & catalyst for showing the community score.